docs/en/custom-elements/introduction.md

```diff
@@ -0,0 +1,164 @@
+
+> index.html
+
+```html
+<html>
+  <head>
+    <link rel="stylesheet" href="ref-email-signup.css">
+  </head>
+  <body>
+    <ref-email-signup></ref-email-signup>
+    <script async src="ref-email-signup.js"></script>
+  </body>
+</html>
+```
+
+| Feature           | Description                                                                                                                                                              |
+| ----------------- | ------------------------------------------------------------------------------------------------------------------------------------------------------------------------ |
+| Widget Properties | Based on the type and name of the property, Dojo will decide if the property should be an attribute, property, or event on the custom element.                           |
+| Child Renderers   | Widgets that use child objects or renderers are supported in custom elements by using slots.                                                                             |
+| Lazy Loading      | The custom element script does not need to be loaded before the custom element is referenced on the page. Elements will be hydrated automatically when the script loads. |
+
+## Building Custom Elements
+
+Dojo widgets can be built as custom elements using @dojo/cli-build-widget. For this example, we're going to use a simple email subscription widget.
+
+> src/widgets/EmailSignup.tsx
+
+```tsx
+import { tsx, create } from '@dojo/framework/core/vdom';
+import * as css from './styles/EmailSignup.m.css';
+import icache from '@dojo/framework/core/middleware/icache';
+
+export interface EmailSignupProperties {
+	title?: string;
+	onSignUp?: () => void;
+	signUpBonusDate?: Date;
+}
+
+const factory = create({ icache }).properties<EmailSignupProperties>();
+
+export default factory(function EmailSignup({ middleware: { icache }, properties }) {
+	const { title, signUpBonusDate } = properties();
+
+	const signedUp = icache.getOrSet('signedUp', false);
+	const signUp = () => {
+		const { onSignUp } = properties();
+
+		// post email address to a server
+		icache.set('signedUp', true);
+		onSignUp && onSignUp();
+	};
+
+	return (
+		<div key="root" classes={css.root}>
+			<div key="title" classes={css.title}>
+				{title || 'Subscribe Now'}
+			</div>
+			{signedUp ? (
+				<div key="confirmation" classes={css.thanks}>
+					Thank you for signing up!
+				</div>
+			) : (
+				<div key="signup" classes={css.signup}>
+					{signUpBonusDate && <div>Sign up by {signUpBonusDate.toLocaleDateString()}</div>}
+					<input type="text" />
+					<button onclick={signUp}>Sign Up</button>
+				</div>
+			)}
+		</div>
+	);
+});
+```
+
+To build a widget as a custom element, add a `build-widget` property to your `.dojorc` file and include the path to any widgets you want to expose as custom elements.
+
+> .dojorc
+
+```json
+{
+	"build-widget": {
+		"prefix": "ref",
```

Comment:
maybe we should explain the prefix and naming strategy below here.

Reply:
Done!